tests/src/main/java/net/neoforged/neoforge/debug/client/TextureAtlasTests.java

```diff
@@ -60,4 +75,64 @@ static void testMaterialAtlas(final DynamicTest test) {
             });
         });
     }
+
+    @TestHolder(description = { "Tests that custom sprite metadata sections get passed through resource reloading properly" }, enabledByDefault = true)
+    @GameTest
+    @EmptyTemplate
+    static void defaultSpriteMetadataSections(final DynamicTest test) {
+        String modId = test.createModId();
+
+        var testResource = ResourceLocation.fromNamespaceAndPath(modId, "resource.png");
+        var sectionType = new MetadataSectionType<>("default_metadata_test", Codec.BOOL);
+
+        var provider = new SpriteSource() {
```

Comment:
Why not use a sprite source provided by vanilla? E.g. the single file one seems sufficient.

Reply:
I wanted to be explicit in the test code, but yes, that'd probably be better.